transaction-view/src/resolved_transaction_view.rs

```diff
@@ -0,0 +1,570 @@
+                is_writable_cache[usize::from(ix.program_id_index)] = false;
+            }
+        }
+
+        is_writable_cache
+    }
+
+    fn num_readonly_accounts(&self) -> usize {
+        usize::from(self.view.total_readonly_lookup_accounts())
+            .wrapping_add(usize::from(self.view.num_readonly_signed_static_accounts()))
+            .wrapping_add(usize::from(
+                self.view.num_readonly_unsigned_static_accounts(),
+            ))
+    }
+
+    fn signature_details(&self) -> TransactionSignatureDetails {
+        // counting the number of pre-processor operations separately
+        let mut num_secp256k1_instruction_signatures: u64 = 0;
+        let mut num_ed25519_instruction_signatures: u64 = 0;
+        for (program_id, instruction) in self.program_instructions_iter() {
+            if secp256k1_program::check_id(program_id) {
+                if let Some(num_verifies) = instruction.data.first() {
+                    num_secp256k1_instruction_signatures =
+                        num_secp256k1_instruction_signatures.wrapping_add(u64::from(*num_verifies));
+                }
+            } else if ed25519_program::check_id(program_id) {
+                if let Some(num_verifies) = instruction.data.first() {
+                    num_ed25519_instruction_signatures =
+                        num_ed25519_instruction_signatures.wrapping_add(u64::from(*num_verifies));
+                }
+            }
+        }
+
+        TransactionSignatureDetails::new(
+            u64::from(self.view.num_required_signatures()),
+            num_secp256k1_instruction_signatures,
+            num_ed25519_instruction_signatures,
+        )
+    }
+}
+
+impl<D: TransactionData> SVMMessage for ResolvedTransactionView<D> {
+    fn num_total_signatures(&self) -> u64 {
+        self.signature_details().total_signatures()
+    }
+
+    fn num_write_locks(&self) -> u64 {
+        self.account_keys()
+            .len()
+            .wrapping_sub(self.num_readonly_accounts()) as u64
+    }
+
+    fn recent_blockhash(&self) -> &Hash {
+        self.view.recent_blockhash()
+    }
+
+    fn num_instructions(&self) -> usize {
+        usize::from(self.view.num_instructions())
+    }
+
+    fn instructions_iter(&self) -> impl Iterator<Item = SVMInstruction> {
+        self.view.instructions_iter()
+    }
+
+    fn program_instructions_iter(
+        &self,
+    ) -> impl Iterator<
+        Item = (
+            &solana_sdk::pubkey::Pubkey,
+            solana_svm_transaction::instruction::SVMInstruction,
+        ),
+    > {
+        self.view.program_instructions_iter()
+    }
+
+    fn account_keys(&self) -> AccountKeys {
+        AccountKeys::new(
+            self.view.static_account_keys(),
+            self.resolved_addresses.as_ref(),
+        )
+    }
+
+    fn fee_payer(&self) -> &Pubkey {
+        &self.view.static_account_keys()[0]
+    }
+
+    fn is_writable(&self, index: usize) -> bool {
+        self.writable_cache[index]
```

Comment:
We should get + unwrap(false) here rather than panicking from a bad caller

Reply:
100% right, thank you! 4c06dc663c2c020004f13ea8dfc23c4d979da231